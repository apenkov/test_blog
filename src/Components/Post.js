import React, {Component} from 'react';

export default class Post extends Component {
  constructor(props) {
    super(props);

    this.state = {
      visible: false,
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState(state => ({
      visible: !this.state.visible
    }))
  }

  render() {
    const post = this.props.post;
    return (
      <div className="post">
        <p className="title">{post.title}</p>
        <p className="preview" style={{visibility: this.state.visible ? 'hidden' : 'visible'}}>{post.littleText}</p>
        <button href="#" className={`post_readMore`} onClick={this.handleClick}
                style={{visibility: this.state.visible ? 'hidden' : 'visible'}}>Читать
        </button>
        <p className={`body`} style={{
          visibility: this.state.visible ? 'visible' : 'hidden',
          height: this.state.visible ? 'auto' : 0,
          transition: 'height 2s ease'
        }}
        dangerouslySetInnerHTML={{__html:post.text}}></p>
        <button href="#" className={`post_readLess`} onClick={this.handleClick}
                style={{visibility: this.state.visible ? 'visible' : 'hidden'}}>Скрыть
        </button>
      </div>
    )
  }
}
