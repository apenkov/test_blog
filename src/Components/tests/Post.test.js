import React from 'react';
import renderer from 'react-test-renderer';
import Post from "../Post";

describe('post tests', () => {
  it('first tests', () => {
    const post = renderer
      .create(<Post data={{title:'title',littleText:'little text', text:'big text'}}/>)
      .toJSON();
    expect(post).toMatchSnapshot();
  })
})
