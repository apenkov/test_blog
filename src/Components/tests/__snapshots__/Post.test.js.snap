// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post tests first tests 1`] = `
<div
  className="post"
>
  <p
    className="title"
  >
    title
  </p>
  <p
    className="preview"
  >
    little text
  </p>
  <button
    className="post_readMore"
    href="#"
    onClick={[Function]}
    style={
      Object {
        "visibility": "visible",
      }
    }
  >
    Читать
  </button>
  <p
    className="body"
    style={
      Object {
        "height": 0,
        "transition": "height 2s ease",
        "visibility": "hidden",
      }
    }
  >
    big text
  </p>
</div>
`;
